src/rockstor/system/osi.py

```diff
@@ -1203,6 +1203,30 @@ def set_system_rtc_wake(wakeup_epoch):
     return None
 
 
+def is_network_device_responding(address):
+    """
+    Small function that sends ICMP echo requests to a given network device,
+    either using the hostname or IP address.
+    :param address: the hostname or IP address of the device to be pinged
+    :return: true if the device is responding, false if not
+    """
+    # because of -c 3, three requests will be sent
+    output, error, return_code = run_command([PING, '-c 3 -q', address], log=False, throw=False)
```

Comment:
Style reformatting + fix for unrecognized command in  up-to-date Tumbleweed:
```suggestion
    cmd = [PING, '-c', 3, '-q', address]
    o, e, rc = run_command(cmd, log=True, throw=False)
```